src/approaches/interactive_learning_approach.py

```diff
@@ -255,6 +258,78 @@ def _create_glib_interaction_strategy(
         termination_function = task.goal_holds
         return act_policy, termination_function
 
+    def _create_greedy_lookahead_interaction_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Sample a certain number of max-length trajectories and pick the one
+        that has the highest cumulative score."""
+        init = self._train_tasks[train_task_idx].init
+        # Create all applicable ground NSRTs
+        ground_nsrts: List[_GroundNSRT] = []
+        for nsrt in sorted(self._get_current_nsrts()):
+            ground_nsrts.extend(utils.all_ground_nsrts(nsrt, list(init)))
+        # Sample trajectories by sampling random sequences of NSRTs.
+        best_score = -np.inf
+        best_options = []
+        for _ in range(CFG.interactive_max_num_trajectories):
+            state = init.copy()
+            options = []
+            trajectory_length = 0
+            total_score = 0.0
+            while trajectory_length < CFG.interactive_max_trajectory_length:
+                # Sample an NSRT that has preconditions satisfied in the
+                # current state.
+                ground_nsrt = self._sample_applicable_ground_nsrt(
+                    state, ground_nsrts)
+                if ground_nsrt is None:  # No applicable NSRTs
+                    break
+                assert all(a.holds for a in ground_nsrt.preconditions)
+                # Sample an option. Note that goal is assumed not used.
+                assert not CFG.sampler_learning_use_goals
+                option = ground_nsrt.sample_option(state,
+                                                   goal=set(),
+                                                   rng=self._rng)
+                # Assume for now that options will be initiable when the
+                # preconditions of the NSRT are satisfied.
+                assert option.initiable(state)
+                state, num_actions = \
+                    self._option_model.get_next_state_and_num_actions(state,
+                                                                      option)
+                # Special case: if the num actions is 0, something went wrong,
+                # and we don't want to use this option after all. To prevent
+                # possible infinite loops, just break immediately in this case.
+                if num_actions == 0:
+                    break
+                options.append(option)
+                trajectory_length += num_actions
+                # Update the total score.
+                atoms = utils.abstract(state, self._predicates_to_learn)
+                total_score += self._score_atom_set(atoms, state)
+            if total_score > best_score:
+                best_score = total_score
+                assert not np.isinf(best_score)
```

Comment:
oh i actually meant that we should do this assert after the loop, right before (and at the same indentation level as) `act_policy = utils.option_plan_to_policy(best_options)`. the idea is to make sure that we saw at least one trajectory, so best_score isn't just equal to its initialization (`-np.inf`)